utils.c

```diff
@@ -594,6 +599,8 @@ load_file_data (int     fd,
     {
       if (data_len == data_read + 1)
         {
+          if (data_len > SIZE_MAX / 2)
+            return NULL;
```

Comment:
This needs to either exit the program with an appropriate fatal error (`die_oom()` seems close enough, or use `die ("File too large to read into memory")` or similar), or set `errno` (perhaps to `ENOMEM`) before returning null. Otherwise, the function will return NULL without setting a correct error.